### PR TITLE
feat(map): add clickable event markers and event-focused navigation

### LINKED ISSUE #176: Add clickable event markers and navigation from events to map

### PR DESCRIPTION
**Pull request by othmanehousni**
> [!WARNING]
> - there is LOTS and LOTS 😨  of tests that were done to handle as much as possible the new implementation and the newly added code. however, some parts are _impossible_ to test so I tried making sure that everything works around it since the e2e test already handles that part
> - some bio tests needed to be removed since they are obsolete and after discussion with their author, we decided it is best just to comment them for now to ensure everything works in the future

image

## **What**

This pull request adds support for **clickable event markers on the map** and introduces **navigation from event cards directly to their map locations**.
It enhances both the **Map** and **Event** experiences by integrating event context into navigation and improving map interactivity.

## **Why**

Previously, the map view displayed events visually but lacked interactive behavior.
Users could not tap on events to focus on their details, nor navigate from an event’s card to its precise location on the map.
This PR bridges those gaps, enabling a seamless transition between the event detail screen and its geographical context.

### **Map Enhancements**

* Added **clickable event markers** on the map.
* Introduced **animated focus** on selected events using smooth camera transitions (`animateToSelectedEvent`).
* Updated `MapViewModel` with new state fields:

  * `selectedEvent`
  * `selectedEventLocation`
  * `shouldAnimateToSelectedEvent`
* Added new `MapViewEvent.SelectEvent` to handle event marker selection.
* Integrated automatic animation when an event is selected.

### **Event Screen Integration**

* Event cards now include a **“View on Map” button**.
* Clicking this button navigates to the map, passing coordinates and the event UID.
* The map highlights and centers on the corresponding event marker.

### **Navigation Updates**

* Modified `Route.mapWithLocation()` to accept an optional `eventUid` parameter.
* Updated `MainActivity` and `Navigation.kt` to handle navigation routes like:

  ```
  map/{latitude}/{longitude}/{zoom}?eventUid={eventUid}
  ```
* Updated `HomeScreen` to correctly handle camera and navigation states.

### **Testing**

* **Unit tests** for navigation routes and new `MapViewModel` event handling.
* **Integration tests** verifying:

  * Event → Map navigation flow
  * Marker selection and zoom behavior
  * Null, invalid, and valid event UID edge cases
* **UI tests** added and updated for:

  * `EventViewTest`
  * `MapScreenTest`
  * `MapViewModelTest`
  * `NavigationTest`

## **Linked Issue**

Closes #176 

## **Expected Behavior**

scc

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/ui/event/EventView.kt [hunk 1]
```diff
@@ -454,7 +454,11 @@ private fun CommonActionButtons(
         id = R.drawable.ic_location_pin,
         onClick = {
           currentEvent.location?.let { location ->
-            val route = Route.mapWithLocation(location.latitude, location.longitude)
+            // Navigate to map with location and event UID to automatically select and display the
+            // event
+            val route =
+                Route.mapWithLocation(
+                    location.latitude, location.longitude, eventUid = currentEvent.uid)
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/event/EventView.kt]: I love your reformatting and the new parameter at the end, well done!
However, since the first two parameters are unnamed, it would make more sense to have the last parameter be unnamed as well 😅

File: app/src/main/java/com/github/se/studentconnect/ui/navigation/Navigation.kt [hunk 2]
```diff
@@ -32,8 +32,14 @@ object Route {
...
-  fun mapWithLocation(latitude: Double, longitude: Double, zoom: Double = 15.0): String {
-    return "map/$latitude/$longitude/$zoom"
+  fun mapWithLocation(
+      latitude: Double,
+      longitude: Double,
+      zoom: Double = 15.0,
+      eventUid: String? = null
+  ): String {
+    val baseRoute = "map/$latitude/$longitude/$zoom"
+    return if (eventUid != null) "$baseRoute?eventUid=$eventUid" else baseRoute
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/navigation/Navigation.kt]: nice use of query parameters

File: app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt [hunk 3]
```diff
@@ -171,6 +174,7 @@ fun HomeScreen(
     onClearScrollTarget: () -> Unit = {}
 ) {
   var showNotifications by remember { mutableStateOf(false) }
+  // var cameraMode by remember { mutableStateOf(CameraMode.QR_SCAN) }
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt]: this commented out code should be deleted

File: app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt [hunk 4]
```diff
@@ -241,26 +298,47 @@ private fun SearchBar(
...
+   * animation to the event location. This prevents the card from being dismissed due to camera
+   * movement during the automatic animation.
+   */
+  val isAnimatingToEvent = remember { mutableStateOf(false) }
+
+  // Set animation flag when event is selected and reset after animation completes
+  LaunchedEffect(selectedEvent) {
+    if (selectedEvent != null) {
+      isAnimatingToEvent.value = true
+      kotlinx.coroutines.delay(1000) // Wait for animation to complete (1 second)
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt]: are you sure it is a good idea to hard code this delay? isn't there a better deterministic way to check that the animation is completed?

File: app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt [hunk 5]
```diff
@@ -280,6 +358,71 @@ private fun MapContainer(
...
+                          // Clicked on map, not on a marker - clear selection
+                          onEventSelected(null)
+                        }
+                      }
+                          ?: run {
+                            // Error querying features - clear selection
+                            onEventSelected(null)
+                          }
+                    }
+                true
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt]: very hard to read due to lots nesting, maybe break out portions into separated variables/blocks of code

File: app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt [hunk 6]
```diff
@@ -424,11 +575,137 @@ private fun BoxScope.MapActionButtons(
...
+                        modifier = Modifier.size(Size.ICON))
+                    Spacer(modifier = Modifier.width(Padding.VERTICAL_SPACING))
+                    Text(
+                        text = "Max: $capacity people",
+                        style = MaterialTheme.typography.bodyMedium,
+                        color = MaterialTheme.colorScheme.onSurface)
+                  }
+                }
+              }
+            }
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt]: good ui code, no ssues

File: app/src/main/java/com/github/se/studentconnect/MainActivity.kt [hunk 7]
```diff
@@ -211,14 +211,30 @@ fun MainContent() {
...
+ *
+ * @param navController The navigation controller for handling navigation between screens
+ * @param selectedTab The currently selected bottom navigation tab
+ * @param onTabSelected Callback invoked when a tab is selected
+ * @param shouldOpenQRScanner Whether the QR scanner should be opened automatically
+ * @param onQRScannerStateChange Callback to notify when QR scanner state changes
+ */
 @Composable
-private fun MainAppContent(
+internal fun MainAppContent(
```
[the remark this answers is not in the document]
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/MainActivity.kt]: thanks! it became easier to test some content when switching to internal containerization

File: app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt (hunk 3 above)
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt]: you are right, im just hoping it wont break the whole thing when merging...

File: app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt [hunk 8]
```diff
@@ -192,6 +196,12 @@ fun HomeScreen(
     }
   }
 
+  // Notify parent when camera mode selector becomes active/inactive to control bottom nav
+  // visibility
+  LaunchedEffect(pagerState.currentPage) {
+    onCameraActiveChange(pagerState.currentPage == HomeScreenConstants.PAGER_SCANNER_PAGE)
```
[the remark this answers is not in the document]
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt]: this was an artefact of another PR that needed to be removed. deleted 👍🏻

File: app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt (hunk 5 above)
Comment (83759) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/map/Map.kt]: okay no probs
